### ISSUE TITLE
Feature Request: Drop cats monadless dependency

### ISSUE THREAD

**Issue by github-louis-fruleux** [coursier/coursier#2473]
The project seems unmaintained, and creates conflicts with newer versions.

monadless seems stuck with cats-core 0.9.0 meanwhile, the current version is cats-core 2.X

**Comment by alexarchambault** [comment]
I agree it should be dropped… Maybe a solution based on `-Xasync`, like in [cats-effect-cps](https://github.com/typelevel/cats-effect-cps) could work as a substitute. I've been wanting to use that kind of thing with `coursier.util.Task`, to clean up some code areas in coursier (in `FileCache` in particular).
